internal/pkg/skuba/addons/dex.go

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package addons
+
+import (
+	"github.com/SUSE/skuba/internal/pkg/skuba/dex"
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	"github.com/SUSE/skuba/internal/pkg/skuba/skuba"
+	skubaconstants "github.com/SUSE/skuba/pkg/skuba"
+	"github.com/pkg/errors"
+	"k8s.io/kubernetes/cmd/kubeadm/app/images"
+)
+
+var gangwayClientSecret string
+
+func init() {
+	registerAddon(kubernetes.Dex, renderDexTemplate, dexCallbacks{}, normalPriority)
+}
+
+func (renderContext renderContext) DexImage() string {
+	return images.GetGenericImage(skubaconstants.ImageRepository, "caasp-dex",
+		kubernetes.AddonVersionForClusterVersion(kubernetes.Dex, renderContext.config.ClusterVersion).Version)
+}
+
+func (renderContext renderContext) GangwayClientSecret() string {
```

Comment:
Cool way, thanks for teach me. Now I know rendering could pass function, not just variable. :1st_place_medal: